psi4/driver/procrouting/proc.py

```diff
@@ -2691,6 +2695,35 @@ def run_bccd(name, **kwargs):
     optstash.restore()
     return ref_wfn
 
+def run_tdscf_energy(**kwargs):
+
+    # Get a wfn in case we aren't given one
+    ref_wfn = kwargs.get('ref_wfn', None)
+    if ref_wfn is None:
+        raise ValidationError("TDSCF: No reference wave function passed!")
+
+    states = core.get_global_option("TDSCF_STATES_PER_IRREP")
+
+    # some sanity checks
+    if sum(states) == 0:
+        raise ValidationError("TDSCF: No states requested in TDSCF_STATES_PER_IRREP") 
+
+    if len(states) != ref_wfn.nirrep():
+        raise ValidationError("TDSCF: Requested TDSCF_STATES_PER_IRREP do not match wave function symmetry") 
+
+    ssuper_name = ref_wfn.functional().name()
+
+    # Do we need this return value? 
+    ret = response.scf_response.tdscf_excitations(ref_wfn, states_per_irrep = states,
+                                                  triplet = core.get_global_option("TDSCF_TRIPLETS"),
+                                                  tda = core.get_global_option("TDSCF_TDA"),
```

Comment:
get_option for the triplets, tda, guess. E/R_tol are still in negotiations, I guess.